students/RoyC/Lesson04/mailroom.py

```diff
@@ -0,0 +1,130 @@
+        if donor_name == "list":
+            for d in donors:
+                print(d)
+            print()
+        else:
+            donations = donors.setdefault(donor_name, [])
+            amt = float(input("Enter amount of donation: "))
+            donations.append(amt)
+            record = {'donor_name' : donor_name, 'amt' : "{:.2f}".format(amt), 'total' : "{:.2f}".format(get_total(donations))}
+            print(ltr_template.format(**record))
+            break
+    
+def thank_all():
+    """
+    Write thank you letters for all donors to separate files
+    """
+    for donor, donations in donors.items():
+        f = open(donor.replace(' ',  '_') + ".txt", 'w')
+        last_donation = donations[-1:][0]
+        donor_record = {'donor_name' : donor, 'amt' : "{:.2f}".format(last_donation), 'total' : "{:.2f}".format(get_total(donations))}
+        f.write(ltr_template.format(**donor_record))
+    print("\n{:d} thank you letters have been written!\n".format(len(donors)))
+
+def get_total(donations):
+    """
+    Return the total of the given list of donations
+    Argument:
+        donations - list of donation amounts (floating point nums)
+    """
+    total = 0;
+    for donation in donations:
+        total += donation
+    return total
+
+def get_avg(donations):
+    """
+    Return the average of the given list of donations
+    Argument:
+        donations - list of donation amounts (floating point nums)
+    """
+    return get_total(donations) / len(donations)
+    
+def total_sort(row_datum):
+    """
+    Sort donors for donor report based on total donations
+    """
+    return row_datum[1]
+
+def create_report():
+    """
+    Print out a report listing historical donor totals to date
+    """
+    # print the report header
+    header_row = "\n\n{:25} | {:11} | {:9} | {:12}".format("Donor Name", "Total Given", "Num Gifts", "Average Gift")
+    print(header_row)
+    print("-" * len(header_row))
+    # create list of row data from donors
+    row_data = []
+    for donor, donations in donors.items():
+        row_data.append([donor, get_total(donations), len(donations), get_avg(donations)])
+    # sort the list of donors (total donations, most to least) before printing report
+    row_data.sort(key=total_sort, reverse=True)
+    # iterate sorted donor list printing formatted rows
+    for row_datum in row_data:
+        print("{:28}${:>10.2f}{:>12}   ${:>12.2f}".format(row_datum[0], row_datum[1], row_datum[2], row_datum[3]))
+    print("\n\n")
+    
+def quit_menu():
+    """
+    Return text to indicate operator selects quit
+    """
+    return "quit selected"
+
+def display_menu(prompt, menu_dict):
+    """
+    Continually display the given options prompt, then prompt for a selection and invoke
+    associated function from menu dict. If function returns exit string, then break loop.
+    """
+    while True:
+        choice = input(prompt)
+        if menu_dict[choice]() == "quit selected":
+            break;
```

Comment:
instead of checking for "quit selected", have the `quit_menu` function use `exit` or `quit` function to handle existing seamlessly.